app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -230,9 +230,12 @@ class HomeFragment : Fragment() {
         setupHomeMenu()
 
         viewLifecycleOwner.lifecycleScope.launch(IO) {
+            // This should be unnecessary, but we've seen crashes around it. See #6832
+            val context = context ?: return@launch
```

Comment:
If I understand correctly, `viewLifecycleOwner.lifecycleScope` should cancel after `onDestroy`, which should be guaranteed to occur prior to `onDetatch`, so it should be impossible for context to ever be null here.  That said, the stack traces from the Sentry events seem pretty clear.  Am I missing something here?  I'd love to get your opinion @pocmo @csadilek @colintheshots

Reply:
We have been seeing issues with `lifecycleScope` and `Activity.recreate()` (which Fenix does for theme and private mode changes etc.). In that situation the scope from the activity that got destroyed never gets cancelled. We found this after forever debugging such a "this cannot happen" situation too. I never got around trying to isolate this in an empty app to file an androidx issue. :(
So, we got rid of `lifecycleScope` in that situation and started using a `view bound` scope, e.g. see here:
https://github.com/mozilla-mobile/android-components/blob/master/components/lib/state/src/main/java/mozilla/components/lib/state/ext/Fragment.kt#L32-L56

Reply:
😱